src/util/createServerRootMixin.js

```diff
@@ -14,13 +15,33 @@ function walkIndex(indexWidget, visit) {
   });
 }
 
-function renderToString(app, _renderToString) {
-  return new Promise((resolve, reject) =>
-    _renderToString(app, (err, res) => {
-      if (err) reject(err);
-      resolve(res);
-    })
-  );
+export function renderToString(app) {
+  let _renderToString;
+  try {
+    _renderToString = isVue3
+      ? require('@vue/server-renderer').renderToString
```

Comment:
this needs to be added to peer dependencies, but with peerDependenciesMeta as optional (https://docs.npmjs.com/cli/v7/configuring-npm/package-json#peerdependenciesmeta) (and also same for vue-server-renderer)

Reply:
https://github.com/algolia/vue-instantsearch/pull/1020/commits/3b264d2b1f950fea76d141d8eac49ae7c3b2f42d